x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java

```diff
@@ -56,20 +56,51 @@
  *         Find a function in this package similar to the one you are working on and copy it to build
  *         yours. There's some ceremony required in each function class to make it constant foldable
  *         and return the right types. Take a stab at these, but don't worry too much about getting
- *         it right.
+ *         it right. Your function might extend from one of several abstract base classes, all of
+ *         those are fine for this guide, but might have special instructions called out later.
+ *         Known good base classes:
+ *         <ul>
+ *             <li>{@link org.elasticsearch.xpack.esql.expression.function.scalar.convert.AbstractConvertFunction}</li>
+ *             <li>{@link org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.AbstractMultivalueFunction}</li>
+ *             <li>{@link org.elasticsearch.xpack.esql.expression.function.scalar.UnaryScalarFunction}
+ *                 or any subclass like {@code AbstractTrigonometricFunction}</li>
+ *             <li>{@link org.elasticsearch.xpack.esql.expression.function.scalar.EsqlScalarFunction}</li>
+ *             <li>{@link org.elasticsearch.xpack.esql.expression.function.scalar.math.DoubleConstantFunction}</li>
+ *         </ul>
  *     </li>
  *     <li>
  *         There are also methods annotated with {@link org.elasticsearch.compute.ann.Evaluator}
- *         that contain the actual inner implementation of the function. Modify those to look right
- *         and click {@code Build->Recompile 'FunctionName.java'} in IntelliJ or run the
- *         {@code CsvTests} again. This should generate an
- *         {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator} implementation
- *         calling the method annotated with {@link org.elasticsearch.compute.ann.Evaluator}. Please commit the
- *         generated evaluator before submitting your PR.
- *     <li>
- *         Once your evaluator is generated you can implement
- *         {@link org.elasticsearch.xpack.esql.evaluator.mapper.EvaluatorMapper#toEvaluator},
- *         having it return the generated evaluator.
+ *         that contain the actual inner implementation of the function. They are usually named
+ *         "process" or "processInts" or "processBar". Modify those to look right and click
+ *         {@code Build->Recompile 'FunctionName.java'} in IntelliJ or run the {@code CsvTests}
+ *         again. This should generate an {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator}
+ *         implementation calling the method annotated with {@link org.elasticsearch.compute.ann.Evaluator}.
```

Comment:
> click Build->Recompile 'FunctionName.java' in IntelliJ or run the  CsvTests again

You actually need to run both (One for IntelliJ to recognize the files, and the other to generate the real files. As IntelliJ and Gradle generate them in different locations:

![image](https://github.com/elastic/elasticsearch/assets/5280509/4db1d938-1006-4bd1-87ff-dc15f6256d2d)

So I would maybe go with something like:
```suggestion
 *         "process" or "processInts" or "processBar". Modify those to look right and run the {@code CsvTests} again. This should generate an {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator}
 *         implementation calling the method annotated with {@link org.elasticsearch.compute.ann.Evaluator}.
 *.        To make it work with IntelliJ, also click {@code Build->Recompile 'FunctionName.java'}.
```

I don't know if we can fix it instead, would be nice. [But I see you tried it already](https://github.com/elastic/elasticsearch/pull/107313/files#diff-53f3a207a751d5038b9c7211709242c1fe2c4668e79d33f6b9ff5b04dcfea446R46)

Reply:
Yeah, if you want to fight with the build system and intellij, be my guest. They do not agree.